include/motor-control/core/linear_motion_system.hpp

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <concepts>
+
+namespace lms {
+
+template <typename MC>
+concept MotorMechanicalConfig = requires(MC mc) {
```

Comment:
I don't get the value of this concept. It's not documented. It doesn't require mm_per_rev to be any particular type.